lib/spring/client/run.rb

```diff
@@ -114,7 +114,16 @@ def stop_server
       end
 
       def verify_server_version
-        server_version = server.gets.chomp
+        unless IO.select([server], [], [], CONNECT_TIMEOUT)
```

Comment:
So note that this only tell us there's something to read, it doesn't guarantee it's a whole line, so we can still potentially get stuck in `server.gets`.

However it should catch a large portion of such issues, so I think it's a good improvement.